snappy_pipeline/workflows/mobile_element_insertion/__init__.py

```diff
@@ -0,0 +1,360 @@
+
+    @staticmethod
+    @dictify
+    def _get_output_files_cluster():
+        """Yield output files' patterns for scramble cluster call."""
+        # Initialise variables
+        name_pattern = "{mapper}.scramble.{library_name}"
+        ext = "txt"
+        # Yield
+        yield ext, "work/{name_pattern}/out/{name_pattern}_cluster.{ext}".format(
+            name_pattern=name_pattern, ext=ext
+        )
+
+    @staticmethod
+    @dictify
+    def _get_output_files_analysis():
+        """Yield output files' patterns for scramble call."""
+        # Initialise variables
+        name_pattern = "{mapper}.scramble.{library_name}"
+        ext_dict = {"txt": "_MEIs.txt", "txt_md5": "_MEIs.txt.md5"}
+        # Yield
+        for key, ext in ext_dict.items():
+            yield key, "work/{name_pattern}/out/{name_pattern}{ext}".format(
+                name_pattern=name_pattern, ext=ext
+            )
+
+    def get_log_file(self, action):
+        """Return log function for scramble rules.
+
+        :param action: Action (i.e., step) in the workflow.
+        :type action: str
+
+        :return: Returns log function for scramble rule based on inputted action.
+
+        :raises UnsupportedActionException: if action not in class defined list of valid actions.
+        """
+        # Initialise variable
+        name_pattern = "{mapper}.scramble.{library_name}"
+
+        # Validate inputted action
+        if action not in self.actions:
+            valid_actions_str = ", ".join(self.actions)
+            error_message = "Action '{action}' is not supported. Valid options: {options}".format(
+                action=action, options=valid_actions_str
+            )
+            raise UnsupportedActionException(error_message)
+        if action == "annotate":
+            name_pattern_annotated = "{mapper}.scramble_annotated.{library_name}"
+            return "work/{name_pattern}/log/{name_pattern}.log".format(
+                name_pattern=name_pattern_annotated
+            )
+        return "work/{name_pattern}/log/{name_pattern}_{action}.log".format(
+            name_pattern=name_pattern, action=action
+        )
+
+    def get_params(self, action):
+        """Get parameters.
+
+        :param action: Action, i.e., step being performed.
+        :type action: str
+
+        :return: Returns method to get files required to run analysis part of scramble.
+        """
+        assert action == "analysis", "Parameters are only available for action 'analysis'."
+        return self._get_analysis_parameters
+
+    def _get_analysis_parameters(self, _wildcards):
+        """Get parameters.
+
+        :param _wildcards: Snakemake rule wildcards (unused).
+        :type _wildcards: snakemake.io.Wildcards
+
+        :return: Returns parameters required to run analysis part of scramble.
+        """
+        # Define required paths
+        base_path = self.config["scramble_install_dir"]
+        rscript_path = os.path.join(base_path, "SCRAMble.R")
```

Comment:
Can we switch to using scrambl package from bioconda?

https://bioconda.github.io/recipes/scramble/README.html?highlight=scramble